src/Components/AuthInput/AuthInput.tsx

```diff
@@ -1,67 +1,68 @@
-import React, { useState } from 'react';
-import { KeyboardTypeOptions, Text, TextInput, View } from 'react-native';
+import { useState, CSSProperties } from 'react';
+import {
+  KeyboardTypeOptions,
+  Text,
+  TextInput,
+  View,
+  ViewBase,
+} from 'react-native';
+import { z } from 'zod';
 
 import styles from './styles';
 
 interface AuthInputProps {
   input: string;
-  setInput: React.Dispatch<React.SetStateAction<string>>;
-  defaultValue: string;
+  onChangeInput: (input: string) => void;
+  labelText: string;
+  placeholderText: string;
   isPassword: boolean;
-  displayInput: boolean;
-  setDisplayInput: React.Dispatch<React.SetStateAction<boolean>>;
   keyboard: KeyboardTypeOptions;
   autoCapitalization: boolean;
-  placeholder: string;
-  setPlaceholder: React.Dispatch<React.SetStateAction<string>>;
+  isHalfWidth?: boolean;
 }
 
 export default function AuthInput({
```

Comment:
I refactored the AuthInput component so it'll use significantly less `useState`s, and hopefully much less complexity. Take a moment to look through this as it requires different props and validates inputs in a different way.